src/pages/components/progress-bar/usage.mdx

```diff
@@ -0,0 +1,504 @@
+
+Labels or helper text should always be placed above, below, or beside the
+progress bar for readibility. Avoid placing text inside or centering text with
+the progress bar. This visual could cause clutter and inconsistency in text
+alignment. Meanwhile, placing the text far away from the progress bar could
+cause confusion in the context.
+
+<DoDontRow>
+  <DoDont caption="Do place text inline with the progress bar.">
+
+![Do place text inline with the progress bar](images/progress-bar-usage-12.png)
+
+  </DoDont>
+  <DoDont type="dont" caption="Don’t embed text into the progress bar.">
+
+![Don’t embed text into the progres bar](images/progress-bar-usage-13.png)
+
+  </DoDont>
+</DoDontRow>
+
+<DoDontRow>
+  <DoDont caption="Do place text close to the progress bar.">
+
+![Do place text close to the progress bar](images/progress-bar-usage-14.png)
+
+  </DoDont>
+  <DoDont type="dont" caption="Don't place text far away from the progress bar.">
+
+![Don't place text far away from the progress bar](images/progress-bar-usage-15.png)
+
+  </DoDont>
+</DoDontRow>
+
+## Content
+
+### Main elements
+
+#### Labels
+
+- Labels inform users what information the progress bar is processing.
+- Keep the label short and concise by limiting it to a single line of text.
+- Labels should never change as the progres bar is loading.
+
+#### Helper text
+
+- Helper text is used to show additional information about the process taking
+  place.
+- Common usecases of helper text for a progress bar can either be a generic
+  phrase, for example “Fetching assets…”, or it can indicate an exact amount of
+  something to be completed, for example "42/256 items".
+- When indicating a loading percentage, the percentage number should count up as
+  the bar progresses.
+
+### Further guidance
+
+For further content guidance, see Carbon's
+[content guidelines]([https://www.carbondesignsystem.com/guidelines/content/overview/]).
+
+## Universal behaviors
+
+### States
+
+There are three states for the progress bar: active, success, and error. These
+states also apply on both indeterminate and determinate progress bars. After the
+process completes successfully or unsuccesfully, a progress bar can either
+remain persistent as confirmation or validation, or it can be automatically
+dismissed depending on what is most suitable for the use case.
+
+<Row>
+<Column colLg={8}>
+
+![Progress bar states.](images/progress-bar-usage-16.png)
+
+</Column>
+</Row>
+
+#### Active
+
+The active loading state indicates that the action is still in progress.
+
+#### Successful
+
+The success loading state indicates that the action completed successfully.
+
+#### Error
+
+The error loading state indicates that the action did not successfully complete.
+The error loading state indicates that the action did not successfully complete.
+If an error occurs, an inline notification or error handling within the form
+should appear. As soon as a process fails, the current visual progress remains
```

Comment:
Add a little more detail here to help with understanding like: 
"When the process fails in determinate bars, the current progress remains visible but indicator bar's progress stops in place." 

I also think indeterminate wouldn't stop incrementally on error but span full width like in the example below. I would add a bit about that state.